haystack/components/rankers/hugging_face_tei.py

```diff
@@ -0,0 +1,250 @@
+
+        :returns: A dict containing the component's initialization parameters.
+        """
+        return default_to_dict(
+            self,
+            url=self.url,
+            top_k=self.top_k,
+            timeout=self.timeout,
+            token=self.token.to_dict() if self.token else None,
+            max_retries=self.max_retries,
+            retry_status_codes=self.retry_status_codes,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "HuggingFaceTEIRanker":
+        """
+        Deserializes the component from a configuration dictionary.
+
+        :param data: Configuration dict produced by `to_dict()`.
+
+        :returns: An initialized `HuggingFaceTEIRanker` instance.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["token"])
+        return default_from_dict(cls, data)
+
+    def _compose_response(
+        self, result: Union[Dict[str, str], List[Dict[str, Any]]], top_k: Optional[int], documents: List[Document]
+    ) -> Dict[str, List[Document]]:
+        """
+        Processes the API response into a structured format.
+
+        :param result: The raw response from the API.
+
+        :returns: A dict with key `documents` mapping to the reranked top documents.
+        """
+        if isinstance(result, dict) and "error" in result:
+            error_type = result.get("error_type", "UnknownError")
+            error_msg = result.get("error", "No additional information.")
+            raise RuntimeError(f"HuggingFaceTEIRanker API call failed ({error_type}): {error_msg}")
+
+        # Ensure we have a list of score dicts
+        if not isinstance(result, list):
+            raise RuntimeError("Unexpected response format from HuggingFaceTEIRanker API.")
+
+        # Determine number of docs to return
+        final_k = min(top_k or self.top_k, len(result))
+
+        # Select and return the top_k documents
+        ranked_docs = []
+        for item in result[:final_k]:
+            index: int = item["index"]
+            doc_copy = copy.copy(documents[index])
+            doc_copy.score = item["score"]
+            ranked_docs.append(doc_copy)
+        return {"documents": ranked_docs}
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self,
+        query: str,
+        documents: List[Document],
+        top_k: Optional[int] = None,
+        truncation_direction: Optional[TruncationDirection] = None,
+    ) -> Dict[str, List[Document]]:
+        """
+        Reranks the provided documents by relevance to the query via the TEI API.
+
+        :param query: The user query string to guide reranking.
+        :param documents: List of `Document` objects to rerank.
+        :param top_k: Optional override for the maximum number of documents to return.
+        :param truncation_direction: If set, enable text truncation in the specified direction.
+
+        :returns: A dict with key `documents` mapping to the reranked top documents.
```

Comment:
If it's not simply a list of reranked documents, could we specify what is "mapping" in this case?

```suggestion
        :returns: A dictionary with the following keys:
            - `documents`: A list of reranked documents.
```